### PR TITLE
pd-ctl: make use of new batch ruledef api

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

#2694, as title, save rules using the new api.

### Check List

Tests

- Manual test

Test save rules with `[{count=3}, {count=0}, {count=0}, {count=3}]`, and `[{count=3}, {count=3}, {count=0}]`.

### Release note

- No release note

### CODE REVIEW

**Comment by disksing** [comment]
It seems that we need to find a way to make delete and set atomic too. 😢

**Comment by xhebox** [comment]
If you want so... It can be done in the next PR of deletion. In that PR, I can introduce a `DELETE` method `/config/rules`. Maybe together with a real `/config/batch` API, supporting `delete` and `add` by a slice of new struct `placement.Batch` or so.

File: tools/pd-ctl/pdctl/command/config_command.go [hunk 1]
```diff
@@ -507,31 +507,30 @@ func putPlacementRulesFunc(cmd *cobra.Command, args []string) {
...
-			b, _ := json.Marshal(r)
-			_, err = doRequest(cmd, rulePrefix, http.MethodPost, WithBody("application/json", bytes.NewBuffer(b)))
-			if err != nil {
-				fmt.Printf("failed to save rule %s/%s: %v\n", r.GroupID, r.ID, err)
-				return
-			}
-			fmt.Printf("saved rule %s/%s\n", r.GroupID, r.ID)
+
+	validOpts := opts[:0]
+	for _, Op := range opts {
```
Comment (JmPotato) [inline tools/pd-ctl/pdctl/command/config_command.go:518]: Maybe using `op` instead of `Op` will be more in line with the Go's official standard?

**Comment by nolouch** [comment]
LGTM

**Comment by xhebox** [comment]
Integration is excepted to fail, there is one PR waiting to merge on TiDB. The failed test seems related `/config/label-property`. Do I need to do anything on this PR?

**Comment by xhebox** [comment]
@ti-srebot /run-all-tests

**Pull request merged**